MNT-23047: exifTool command defined as properties and dependant of ru…

vrmoreira opened the pull request:
…nning OS

vrmoreira commented:
Some tests are still failing, therefore converting PR to a draft one

vrmoreira commented:
Testes are green but have a few questions on this:
1. there are no Windows tests for ATS (it was tested manually)
2. When running IT against a Windows machine, they fail. IMO, it's weird

jamalkm commented on alfresco-transform-core-aio/alfresco-transform-core-aio-boot/src/main/java/org/alfresco/transformer/AIOCustomConfig.java:
```diff
@@ -88,6 +91,24 @@
+    @Value("${transform.core.tika.exifTool.unixOS}")
+    private String exifToolCommandOnUnix;
+
+    @Bean("exifTool")
```
I don't see this bean is used anywhere.

vrmoreira replied:
"exifTool" removed. Thanks for pointing that out!

jamalkm commented on pom.xml:
```diff
@@ -248,6 +248,11 @@
             </dependency>
+            <dependency>
+                <groupId>org.yaml</groupId>
+                <artifactId>snakeyaml</artifactId>
```
Why we need this dependency?

vrmoreira replied:
During the development of this PR, Veracode triggered a problem with that dependency, which didn't allowed to proceed the Travis build.

Meanwhile, I've implemented the same change on master, using [ACS-3492](https://alfresco.atlassian.net/browse/ACS-3492).

jamalkm replied:
Where we get this dependency? If we already have snakeyaml in the transform-core then we should just upgrade it.

jamalkm reviewed (commented):
I'm not very familiar with ATS code, but the changes look OK to me, however I left a couple of comments.

jamalkm reviewed (approved):
I'm on PTO until Sep 19th, so will not block the merge.

vrmoreira commented:
@alandavis ,it is possible you to review this PR, to take a little break on the HxP fork? You've got an keep eye to spot defects :-)

vrmoreira commented:
@alandavis , as requested (on Slack), I've fixed the following situations:
* `snakeyaml` was duplicated with the merge (fixed)
* as you said, that clone of the `InputStream` was weird and it was right: `parse` doesn't consume the inputstream, therefore makes no sense to clone it. Fixed.

Can you have another go on the review? Thanks!

alandavis commented on engines/tika/src/main/java/org/alfresco/transform/tika/metadata/extractors/IPTCMetadataExtractor.java:
```diff
@@ -55,6 +56,12 @@ public class IPTCMetadataExtractor extends AbstractTikaMetadataExtractorEmbeddor
 
     private ExifToolParser parser;
+    private RuntimeExec exifRuntimeExec;
+
```
Are you sure this constructor is being called when SpringBoot is creating @Components?

vrmoreira replied:
Removed the default constructor.

alandavis commented on engines/tika/src/main/java/org/alfresco/transform/tika/parsers/ExifToolParser.java:
```diff
@@ -76,13 +77,34 @@ public class ExifToolParser extends ExternalParser {
+                    logger.debug("Command to be executed determined from RuntimeExec");
+                    commandToBeExecuted = exifRuntimeExec.getCommand();
+                }
+                // give some last chance
```
Please replace this comment with a blank line. It add nothing to the understanding and I was left wondering what the 'last chance' was.

alandavis commented on engines/tika/src/main/java/org/alfresco/transform/tika/parsers/ExifToolParser.java:
```diff
@@ -76,13 +77,34 @@ public class ExifToolParser extends ExternalParser {
                 ExternalParser eParser = eParsers.get(0);
-                this.setCommand(eParser.getCommand());
+
+                String[] commandToBeExecuted;
```
Please keep to the style of the class. We already have a mix in this project and adding: "if(" rather than: "if (" adds another. There is a second example of this on line 100.

vrmoreira replied:
Think I didn't miss anything :)

alandavis reviewed (changes requested):
See comments and fix minor changes.

Are you sure the new constructor is automatically being called? There are no tests which is a bit of a worry.

Good to see you found the RuntimeExec class for different OS command lines.

vrmoreira commented on engines/tika/src/main/java/org/alfresco/transform/tika/metadata/extractors/IPTCMetadataExtractor.java:
```diff
@@ -55,6 +56,12 @@ public class IPTCMetadataExtractor extends AbstractTikaMetadataExtractorEmbeddor
     private ExifToolParser parser;
+    private RuntimeExec exifRuntimeExec;
+
+    public IPTCMetadataExtractor(RuntimeExec exifRuntimeExec) {
```
The `IPTCMetadataExtractorTest` uses this constructor and since it's the only constructor, think we've covered.
BTW, the parameter can be null (`ExifToolParser` will deal with nulls).

alandavis replied:
Now the other constructor has been removed and the test class changed to use this one, I think we are fine

alandavis reviewed (approved):
LGTM
